src/game-state/game-data.ts

```diff
@@ -0,0 +1,74 @@
+import { DataModal } from "@data";
+import { SCENE_NAME_START } from "@constants";
+
+export class GameData {
```

Comment:
Can be shorten to
```
constructor(
    
    public data?: null | DataModal
    /* Canvas States */
    public canvas: null | HTMLCanvasElement
    public width: null | number
    public height: null | number
    public canavsElement = null;
    public context?: null | CanvasRenderingContext2D;
    ... // so on

) { }
```

if file length is a concern.

Reply:
For this part I am unsure. The constructor must remain empty because we are initializing this to a singletonInstance of gameStateService which doesn't have any data in the first place. The data or values will come once `gameStateService.setDefaultGameStateValues(data: DataModal,
        canvas: HTMLCanvasElement,
        canavsElement: HTMLCanvasElement)` has been called in the scene-handler.ts.